kratos/elements/levelset_convection_element_simplex.h

```diff
@@ -110,6 +110,21 @@ class LevelSetConvectionElementSimplex
         KRATOS_CATCH("");
     }
 
+    void Initialize(const ProcessInfo& rProcessInfo) override
+    {
+        std::size_t count = 0;
+        for (const auto& r_node : GetGeometry()) {
+            if(r_node.Has(TAU)){
+                count++;
+            }
+        }
+        if (count == TNumNodes) {
+            mElementTauNodal = true;
+        } else {
+            mElementTauNodal = false;
+        }
```

Comment:
Assuming `TNumNodes` is equal to the number of nodes in `GetGeometry()`:

```suggestion
        const auto& r_geometry = GetGeometry();
        mElementTauNodal = std::all_of(r_geometry.begin(),
                                       r_geometry.end(),
                                       [](const auto& rNode) {
                                           rNode.Has(TAU)
                                       });
```

If not, you can use [std::count_if](https://en.cppreference.com/w/cpp/algorithm/count) instead.

Reply:
P.S. the failures could indicate that `mElementTauNodal` is not computed correctly, which might mean that `TNumNodes < GetGeometry().size()` in some cases. Can you please add a debug check in the constructor that makes sure that `TNumNodes == GetGeometry.size()` if they're supposed to be equal?

Reply:
I think we can safely assume that `TNumNodes` equals the geometry size. Note that this is specified at the specialization level and that in the case of a wrong specialization sneaking in, this would result in a segfault when calculating the local system contributions of the element.

Having said this, I'd directly add your suggestion as it is. BTW, I wasn't aware of this `std::all_of`, pretty nice for this kind of tests (which we do a lot in the standard and unfitted fluid). Thanks for the hint!